clone-project.sh

```diff
@@ -1,21 +1,29 @@
-
 slug=$1
 sha=$2
+projdir=$(echo ${slug} | cut -d'-' -f1 | tr . /)
```

Comment:
Yes you are right, I will remove that line but i need the original slug to clone the project. Thus, can I pass ${modifiedslug_with_sha}=${modified_module} and slug seperately to clone-project.sh?

Reply:
You have slug and sha variable and two scenarios:
(1) A project does not exist in input containers - in this scenario just use slug and sha as the slug and sha
(2) A project does exist in input containers - in this scenario just use slug and sha as the slug and ${modifiedslug_with_sha}=${modified_module}

Would this work?

Reply:
I think it wouldn't because for example:
slug: codingchili/excelastic
sha: 6bb7884b1e70f9e588d75bdaf132a84f6e896a7a and I am checking the existance of the project in clone-project.sh. So if the project exists, then I can not create ${modifiedslug_with_sha}=${modified_module} with these two parameter. Am I missing something :(

Reply:
We need not create ${modifiedslug_with_sha}=${modified_module} in clone-project.sh.

For (2), you can just pass in ${modifiedslug_with_sha}=${modified_module} instead of passing in sha. We can then have if statements to check whether sha is actually sha or if it is ${modifiedslug_with_sha}=${modified_module} instead. 

(In fact, sha should be renamed then to sha_or_modified_slug_sha_module)

Reply:
But in this version of sh files, I don't know if the project exists or not in the outside of the clone-project.sh. In run_surefire_rule_azure.sh, I am just giving two parameters in this version; I am not checking if the file is exist or not in run_surefire_rule_azure.sh

Reply:
Ah I see. Then I suggest two options: 
(1) Keep projdir but rename it to be modified_slug_sha_module and just pass in ${modifiedslug_with_sha}=${modified_module} whenever clone-project.sh is called, or
(2) Remove projdir, rename sha to be modified_slug_sha_module, and pas in ${modifiedslug_with_sha}=${modified_module}. You should then be able to retrieve the sha with the following. sha=$(echo $modified_slug_sha_module | rev | cut -d'=' -f2 | cut -d'-' -f1 | rev). Note that this command will give you the short sha but all in intents and purposes, the short sha will function the same as the long sha.